### ISSUE TITLE
Remove unused member false negative

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn#48353]
**Version Used**: 

master

**Steps to Reproduce**:

```csharp
public class C
{
    private const string x = "x";
    private const string y = nameof(y);
}
```

**Expected Behavior**:

Warning on both `x` and `y`

**Actual Behavior**:

Warning on `x` only.

**Comment by CyrusNajmabadi** [comment]
This is potentially out of scope for what the analysis is capable of. similar to how it does not trigger here:

```c#
        private void Foo()
        {
            Foo();
        }
```

or

```c#
        private void Foo() => Bar();
        private void Bar() => Foo();
```

The diagnostic is around "is this unused" not "can this be removed without any effect on the program".  The latter is much more complex analysis and likely in the domain of another feature entirely.

**Comment by Youssef1313** [comment]
Thanks. Closing as by design.
